src/model/gateway.rs

```diff
@@ -1,6 +1,7 @@
 //! Models pertaining to the gateway.
 
 use bitflags::bitflags;
+use reqwest::Url;
```

Comment:
This appears to be a fragile import. While `reqwest` conveniently re-exports the type for us, it's also likely for its `url` and our `url` to diverge. I am quite certain the reason this code works in its current state is due to cargo re-using the same `url` for both `reqwest` and us (that is, only the recent version of `url` is compiled and used across `reqwest` and Serenity). This should be changed to import from the `url` crate directly.
```suggestion
use url::Url;
```